### ISSUE TITLE
[Bug]: Text index not found when using text match on Milvus 2.5 beta

### ISSUE THREAD

**Issue by AreelKhan** [milvus-io/pymilvus#2398]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### TLDR: text index not found when using text match during client.search on Milvus 2.5 beta

I followed the steps outlined [here](https://milvus.io/docs/keyword-match.md), to create a collection with a field with text matching and analyzing turned. This should trigger "...the creation of an inverted index...".

However when I run search with a filter using TEXT_MATCH i get an error saying the text index was not found. When i list the indexes in the collection i only see the one for my vector field. I dropped and recreated the collection to no avail.

Here is the error:
```
MilvusException: <MilvusException: (code=65535, message=fail to search on QueryNode 2: worker(2) query failed: Operator::GetOutput failed for [Operator:PhyFilterBitsNode, plan node id: 32] : Assert "iter != text_indexes_.end()"  => failed to get text index, text index not found at /workspace/source/internal/core/src/segcore/SegmentInterface.cpp:399
```

### Expected Behavior

When i create the collection and call .search() with a filter using TEXT_MATCH, it should leverage the inverted index and filter on documents that satisfy my filter.

### Steps/Code To Reproduce behavior

```markdown
Ive been running the following code inside a notebook with milvus 2.5 beta running using the milvus docker compose file.


from pymilvus import (
    MilvusClient,
    DataType
)


dim = 10
host = "localhost"
port = 19530
collection_name = "temporary"
client = MilvusClient(uri=f"http://{host}:{port}")


schema = client.create_schema(
     auto_id=False,
     enable_dynamic_field=True,
)

schema.add_field(
    field_name="id",
    datatype=DataType.VARCHAR,
    max_length=100,
    is_primary=True,
)

schema.add_field(
    field_name="vector",
    datatype=DataType.FLOAT_VECTOR,
    dim=dim
)

analyzer_params={
    "type": "english"
}

schema.add_field(
    field_name='document',
    datatype=DataType.VARCHAR,
    max_length=200,
    enable_analyzer=True,
    analyzer_params=analyzer_params,
    enable_match=True,
)

index_params = client.prepare_index_params()
index_params.add_index(
    field_name="vector", index_type="AUTOINDEX", metric_type="COSINE"
)

client.create_collection(
    collection_name=collection_name,
    schema=schema,
    index_params=index_params,
)

vector = list(range(0,10,1))
data = [{
    "id": "1",
    "vector": vector,
    "document": "apple",
}]

client.insert(collection_name, data=data)

client.search( # this is where the error happens
    collection_name=collection_name,
    anns_field="vector",
    data=[vector],
    search_params={"params": {"nprobe": 10}},
    limit=10,
    filter = "TEXT_MATCH(document, 'apple')",
    output_fields=["id", "document"]
)

client.list_indexes(collection_name=collection_name)
```

### Environment details

```markdown
- Hardware/Softward conditions (OS, CPU, GPU, Memory): Ubuntu 24.04.1 LTS, Core i9-13900K, NVIDIA RTX 3060, 64 GB RAM
- Method of installation (Docker, or from source): Docker compose (i changed the `DOCKER_VOLUME_DIRECTORY` environment variable)
- Milvus version (v0.3.1, or v0.4.0): v2.5 beta
- Milvus configuration (Settings you made in `server_config.yaml`): No changes, I dont even have a server_config.yaml file
```

**Comment by AreelKhan** [comment]
SOLVED: My dockerfile updated pymilvus via pip, but i forgot to do the same in my local dev set up.
